src/components/TreebankService/TreebankService.js

```diff
@@ -0,0 +1,67 @@
+import React, { Component } from 'react';
+import { Redirect } from 'react-router-dom';
+import {
+  MessagingService,
+  ResponseMessage,
+  WindowIframeDestination as Destination,
+} from 'alpheios-messaging/dist/dev/alpheios-messaging';
+
+const config = {
+  name: 'treebank',
+  targetIframeID: 'string-not-used',
+  targetURL: 'string-not-used',
+};
+
+const error = (request, message) => ResponseMessage.Error(request, new Error(message));
+
+class TreebankService extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = { redirectTo: null };
+
+    this.service = new MessagingService('treebank-service', new Destination(config));
+
+    this.messageHandler = this.messageHandler.bind(this);
+  }
+
+  componentDidMount() {
+    const { name } = config;
+
+    // TODO this line is causing the error (even with try { }):
+    // TypeError: Cannot set property 'origin' of undefined
+    this.service.registerReceiverCallback(name, this.messageHandler);
```

Comment:
It seems like this line is causing the error in both this PR and in https://github.com/perseids-publications/treebank-template/pull/15. When I comment out the line, the error stops happening, but when I wrap the code in a `try { } catch { }` it still occurs.

Even though the error is being thrown, the code still seems to work. But I'd like to figure it out if possible before it's merged into production.

```
TypeError: Cannot set property 'origin' of undefined
WindowIframeDestination._requestHandler
src/destinations/window-iframe-destination.js:89

  86 |  _requestHandler (callbackFn, event) {
  87 |    // `data` prop of an event contains a request message object
  88 |    let request = event.data // eslint-disable-line prefer-const
> 89 |    request.header.origin = event.origin
  90 |       callbackFn(request, this.sendResponse.bind(this))
  91 |  }
  92 |
```

Reply:
This was caused by a message posted by the react-dev-tools. I've entered alpheios-project/messaging#5 to make the messaging code more robust.